craft_parts/executor/part_handler.py

```diff
@@ -483,7 +483,8 @@ def _run_step(
                 step=step_info.step,
                 work_dir=work_dir,
             )
-            return StepContents()
+            files, dirs = _get_files(work_dir)
```

Comment:
I'm not opposed to this behavior "conceptually" but this will include files and directories that were primed by other parts too, right? Is that the current behavior?

Reply:
You mean `craftctl` also needs to get all files list?

Reply:
It sounds better to have it on the server side of the control API where the built-in action is executed. Maybe the step handler could hold the results to be recovered after the client execution?